### PR TITLE
Snowflake pandas

### PR DESCRIPTION
**Pull request by fdosani**
Tagging @jfyu also

Would love to have a bit of a discussion about this PR before taking any action to see if folks align on the approach here. I've overridden the `to_pandas` function for Snowflake. If no `size` is provided it will use the internal snowflake function to get the pandas df. If size is provided to will use the existing inherited `database` functionality.

I was reluctant to change to the `insert_dataframe_to_table` to use `write_pandas()`. The 2 are slightly different in their approach. Our method is not using COPY like the internal snowflake method but inserting directly via a `INSERT` statement, and provides table creation and schema abilities. I updated the docs to explain this. I think if people are really keep on using `write_pandas` they should be able to call it directly from the `cursor`.

Any thoughts on the above?

### CODE REVIEW

**Comment by dasjyao** [comment]
Instead of replacing, I think it might be a better idea to combine `insert_dataframe_to_table` and `write_pandas()` . More specifically, we should keep the functionality of table creation, but replace the `INSERT` statements to use `write_pandas()`. Also, we need to do some performance tests to see how `write_pandas()` improves the speed.

**Comment by jfyu** [comment]
Whether to use `COPY` or `INSERT` I think depends on the performance. But I really like the create table feature because that's a very big use case. 

It does look like `snowflake-connector-python` has an insertion method as well (https://docs.snowflake.com/en/user-guide/python-connector-api.html#pd_writer) but only for `SQLAlchemy`

File: locopy/snowflake.py [hunk 1]
```diff
@@ -439,3 +445,24 @@ def insert_dataframe_to_table(
         logger.info("Inserting records...")
         self.execute(insert_query, params=to_insert, many=True)
         logger.info("Table insertion has completed")
+
+    def to_dataframe(self, size=None):
+        """Return a dataframe of the last query results. This is just a convenience method. This
+        method overrides the base classes implementation in favour for the snowflake connectors
+        built-in ``fetch_pandas_all`` when ``size==None``
```
Comment (jborchma) [inline locopy/snowflake.py:452]: might be silly, but maybe spell out explicitly that when size != None then the inherited method will be used

File: locopy/snowflake.py [hunk 2]
```diff
@@ -371,7 +371,13 @@ def insert_dataframe_to_table(
         self, dataframe, table_name, columns=None, create=False, metadata=None
     ):
         """
-        Insert a Pandas dataframe to an existing table or a new table.
+        Insert a Pandas dataframe to an existing table or a new table. In newer versions of the
```
[the remark this answers is not in the document]
Comment (fdosani) [reply locopy/snowflake.py:374]: 👍  done

File: locopy/snowflake.py (hunk 1 above)
Comment (fdosani) [reply locopy/snowflake.py:452]: 👍

**Comment by fdosani** [comment]
There is a bigger question about the `insert_dataframe_to_table` method. Should we add a flag like `use_snowflake_copy` which will use the underlying snowflake feature if selected vs our version? It should be simple enough and can do it in the next PR if we all agree on that?

**Pull request merged**